src/api/prop.js

```diff
@@ -26,9 +26,10 @@ export const boolean = create({
 });
 
 export const number = create({
-  coerce: alwaysUndefinedIfEmptyOrNumber,
-  deserialize: alwaysUndefinedIfEmptyOrNumber,
-  serialize: alwaysUndefinedIfEmptyOrNumber
+  default: 0,
+  coerce: alwaysZeroIfNotANumberOrNumber,
```

Comment:
Chatted with @treshugart and we're going to get some extra opinions on whether passing `test` should use the default value (`0`) or use `NaN`. Easy code change either way, just need to decide.